k2/csrc/weights.cc

```diff
@@ -33,7 +33,7 @@ void ComputeForwardMaxWeights(const Fsa &fsa, const float *arc_weights,
   int32_t num_states = fsa.NumStates();
   std::fill_n(state_weights, num_states, kDoubleNegativeInfinity);
 
-  const auto &arcs = fsa.data;
+  const auto &arcs = fsa.data + fsa.indexes[0];
```

Comment:
note that we need add `fsa.indexes[0]` here as it may be not `0`. Same changes in those code related with `weights` or `arc_map` (`rmepsilon/determinize/connect/...`)